osu.Game/Screens/Footer/ScreenFooterButton.cs

```diff
@@ -69,94 +69,109 @@ protected LocalisableString Text
         private readonly Box glowBox;
         private readonly Box flashLayer;
 
-        public ScreenFooterButton()
+        public readonly Container TopLevelContent;
+        public readonly OverlayContainer? Overlay;
+
+        public ScreenFooterButton(OverlayContainer? overlay = null)
         {
+            Overlay = overlay;
+
             Size = new Vector2(BUTTON_WIDTH, BUTTON_HEIGHT);
 
-            Child = new Container
+            Child = TopLevelContent = new Container
```

Comment:
I moved the animations inside of the button class as mentioned, and I've replaced `TopLevelContent` with just `Content` (which points to a local container since the button class inherits from `ClickableContainer`).

I'm not entirely sure if commenting is necessary though, the button class doesn't apply transitions on `this` anywhere else, and there's a precedent for this in `ShearedButton`, where `Content` is used to rescale the button instead of `this`, so that if there's a button next to this one all inside of a flow container, it won't move from its place.